src/lightning/mod.rs

```diff
@@ -0,0 +1,316 @@
+            max_inbound_amount.sats,
+            lsp_min_fee_amount.sats,
+            &self.support.get_exchange_rate(),
+            &self.config.receive_limits_config,
+        ))
+    }
+
+    /// Calculate the actual LSP fee for the given amount of an incoming payment.
+    /// If the already existing inbound capacity is enough, no new channel is required.
+    ///
+    /// Parameters:
+    /// * `amount_sat` - amount in sats to compute LSP fee for
+    ///
+    /// For the returned fees to be guaranteed to be accurate, the returned `lsp_fee_params` must be
+    /// provided to [`Bolt11::create`]
+    ///
+    /// Requires network: **yes**
+    pub fn calculate_lsp_fee_for_amount(&self, amount_sat: u64) -> Result<CalculateLspFeeResponse> {
+        let req = OpenChannelFeeRequest {
+            amount_msat: Some(amount_sat.as_sats().msats),
+            expiry: None,
+        };
+        let res = self
+            .rt_handle
+            .block_on(self.sdk.open_channel_fee(req))
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Failed to compute opening channel fee",
+            )?;
+        Ok(CalculateLspFeeResponse {
+            lsp_fee: res
+                .fee_msat
+                .ok_or_permanent_failure("Breez SDK open_channel_fee returned None lsp fee when provided with Some(amount_msat)")?
+                .as_msats()
+                .to_amount_up(&self.support.get_exchange_rate()),
+            lsp_fee_params: Some(res.fee_params),
+        })
+    }
+
+    /// When *receiving* payments, a new channel MAY be required. A fee will be charged to the user.
+    /// This does NOT impact *sending* payments.
+    /// Get information about the fee charged by the LSP for opening new channels
+    ///
+    /// Requires network: **no**
+    pub fn get_lsp_fee(&self) -> Result<LspFee> {
+        let exchange_rate = self.support.get_exchange_rate();
+        let lsp_fee = self.task_manager.lock_unwrap().get_lsp_fee()?;
+        Ok(LspFee {
+            channel_minimum_fee: lsp_fee.min_msat.as_msats().to_amount_up(&exchange_rate),
+            channel_fee_permyriad: lsp_fee.proportional as u64 / 100,
+        })
+    }
+}
+
+fn get_payment_max_routing_fee_mode(
+    config: &MaxRoutingFeeConfig,
+    amount_sat: u64,
+    exchange_rate: &Option<ExchangeRate>,
+) -> MaxRoutingFeeMode {
+    let max_fee_permyriad = Permyriad(config.max_routing_fee_permyriad);
+    let relative_fee = max_fee_permyriad.of(&amount_sat.as_sats());
+    if relative_fee.msats < config.max_routing_fee_exempt_fee_sats.as_sats().msats {
+        MaxRoutingFeeMode::Absolute {
+            max_fee_amount: config
+                .max_routing_fee_exempt_fee_sats
+                .as_sats()
+                .to_amount_up(exchange_rate),
+        }
+    } else {
+        MaxRoutingFeeMode::Relative {
+            max_fee_permyriad: max_fee_permyriad.0,
+        }
+    }
+}
+
+trait Payments {
```

Comment:
Please have a look at the changes introduced here in #1263. It still uses the trait, but it has been considerably simplified because there is a single field to access.

Reply:
Right.. 🤦

Reply:
I'll fix it in the new PR.